src/bin/Interpret.cc

```diff
@@ -1300,4 +1301,25 @@ void Interpret::initializeLogic(opensmt::Logic_t logicType) {
     logic.reset(opensmt::LogicFactory::getInstance(logicType));
 }
 
+std::unique_ptr<MainSolver> Interpret::createMainSolver(SMTConfig & config, const char*  logic_name) {
+
+    if (config.sat_split_type() == spt_none) {
+        return std::make_unique<MainSolver>(*logic, config, std::string(logic_name) + " solver");
+    }
+    else {
+        auto th = MainSolver::createTheory(*logic, config);
+        auto tm = std::unique_ptr<TermMapper>(new TermMapper(*logic));
+        auto thandler = new THandler(*th, *tm);
```

Comment:
Unify the creation of `TermMapper` and `THandler`. Probably use `std::make_unique`?

Reply:
I'm not sure about THandler, we are using it in two  place here:
Moving ownership to MainSolver, and at the same time sending a reference to createInnerSolver

Reply:
Right, we would still need to store a reference to `THandler` before we move the `unique_ptr`.  I would like to re-think this whole mechanism, but for now, it can be kept as is. Except you can still use `std::make_unique` for `TermMapper`.

Reply:
Yes, I did for the TermMapper